src/term/transform/encode_lists.rs

```diff
@@ -82,6 +82,7 @@ impl Term {
 impl Pattern {
   pub fn encode_lists(&mut self) -> bool {
     match self {
+      Pattern::Var(Some(Name(nam))) => nam == LNIL,
```

Comment:
Can you add a TODO here that this should be removed once we separate adding the constructors to the book from desugaring list terms?
This is necessary to work right now, but it's not the right way we should be doing it.